### PR TITLE
update UniRef

### PR DESCRIPTION
**Pull request by LeonardoGonzales**
- add cluster solr search field and enable it in the search
- UniRefEntryLight  model change: commonOrganism id and name moved to Organism 

- add retry logic to datastore and solr index to try to avoid 11PM errors

### CODE REVIEW

File: core-domain/src/main/java/org/uniprot/core/uniref/impl/UniRefEntryLightImpl.java [hunk 1]
```diff
@@ -186,12 +171,10 @@ public boolean equals(Object o) {
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
         UniRefEntryLightImpl that = (UniRefEntryLightImpl) o;
-        return getCommonTaxonId() == that.getCommonTaxonId()
-                && getSequenceLength() == that.getSequenceLength()
+        return getSequenceLength() == that.getSequenceLength()
                 && getMemberCount() == that.getMemberCount()
```
Comment (ahmadshadab) [inline core-domain/src/main/java/org/uniprot/core/uniref/impl/UniRefEntryLightImpl.java:173]: I think, we should include field commonTaxon in equals and hashcode.

File: core-domain/src/test/java/org/uniprot/core/uniref/UniRefUtilsTest.java [hunk 2]
```diff
@@ -29,45 +31,85 @@ void getsTrEMBLIdType() {
...
+    void canGetOrganismCommonNameWithoutCommon() {
         String organism = "scientific only";
-        String cleanedOrganism = UniRefUtils.getOrganismWithoutCommonName(organism);
-        assertNotNull(cleanedOrganism);
-        assertEquals(organism, cleanedOrganism);
+        String scientific = UniRefUtils.getOrganismCommonName(organism);
+        assertNotNull(scientific);
+        assertEquals("", scientific);
     }
 }
```
Comment (ahmadshadab) [inline core-domain/src/test/java/org/uniprot/core/uniref/UniRefUtilsTest.java:108]: good test coverage

File: core-domain/src/test/java/org/uniprot/core/uniref/impl/UniRefEntryLightImplTest.java [hunk 3]
```diff
@@ -16,7 +16,6 @@ void needDefaultConstructorForJsonDeserialization() {
         assertNotNull(obj);
```
Comment (ahmadshadab) [inline core-domain/src/test/java/org/uniprot/core/uniref/impl/UniRefEntryLightImplTest.java:16]: It is better if we have few more tests like verifying the value of commonTaxon and organisms

File: core-parser/src/main/java/org/uniprot/core/parser/tsv/uniref/UniRefEntryLightValueMapper.java [hunk 4]
```diff
@@ -20,8 +20,8 @@
         Map<String, String> map = new HashMap<>();
         map.put(UNIREF_FIELDS.get(0), entry.getId().getValue());
         map.put(UNIREF_FIELDS.get(1), entry.getName());
-        map.put(UNIREF_FIELDS.get(2), entry.getCommonTaxon());
-        map.put(UNIREF_FIELDS.get(3), Long.toString(entry.getCommonTaxonId()));
+        map.put(UNIREF_FIELDS.get(2), EntryMapUtil.convertOrganism(entry.getCommonTaxon()));
+        map.put(UNIREF_FIELDS.get(3), Long.toString(entry.getCommonTaxon().getTaxonId()));
         map.put(UNIREF_FIELDS.get(4), Integer.toString(entry.getMemberCount()));
```
Comment (ahmadshadab) [inline core-parser/src/main/java/org/uniprot/core/parser/tsv/uniref/UniRefEntryLightValueMapper.java:25]: Can getCommonTaxon throw NPE?

File: core-domain/src/main/java/org/uniprot/core/uniref/impl/UniRefEntryLightImpl.java (hunk 1 above)
Comment (LeonardoGonzales) [reply core-domain/src/main/java/org/uniprot/core/uniref/impl/UniRefEntryLightImpl.java:173]: we have it in line 183 and 200

File: core-domain/src/test/java/org/uniprot/core/uniref/UniRefUtilsTest.java (hunk 2 above)
Comment (LeonardoGonzales) [reply core-domain/src/test/java/org/uniprot/core/uniref/UniRefUtilsTest.java:108]: tks

File: core-domain/src/test/java/org/uniprot/core/uniref/impl/UniRefEntryLightImplTest.java (hunk 3 above)
Comment (LeonardoGonzales) [reply core-domain/src/test/java/org/uniprot/core/uniref/impl/UniRefEntryLightImplTest.java:16]: We cover it in the builder test

File: core-parser/src/main/java/org/uniprot/core/parser/tsv/uniref/UniRefEntryLightValueMapper.java (hunk 4 above)
Comment (LeonardoGonzales) [reply core-parser/src/main/java/org/uniprot/core/parser/tsv/uniref/UniRefEntryLightValueMapper.java:31]: I improved this logic, good catch.. tks

File: core-domain/src/main/java/org/uniprot/core/uniref/impl/UniRefEntryLightImpl.java (hunk 1 above)
Comment (ahmadshadab) [reply core-domain/src/main/java/org/uniprot/core/uniref/impl/UniRefEntryLightImpl.java:173]: oh ok. I didnt see. sorry

**Pull request merged**